contracts/near/eth-prover/src/tests_unlock.rs

```diff
@@ -0,0 +1,289 @@
+#[cfg(test)]
+mod tests_unlock {
+    use crate::EthProver;
+    use borsh::BorshSerialize;
+    use eth_types::H256;
+    use hex::FromHex;
+    use near_sdk::PromiseOrValue;
+    use rlp::Rlp;
+    use serde::{Deserialize, Deserializer};
+
+    #[derive(Debug)]
+    struct Hex(pub Vec<u8>);
+
+    impl<'de> Deserialize<'de> for Hex {
+        fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
+        where
+            D: Deserializer<'de>,
+        {
+            let mut s = <String as Deserialize>::deserialize(deserializer)?;
+            if s.starts_with("0x") {
+                s = s[2..].to_string();
+            }
+            if s.len() % 2 == 1 {
+                s.insert_str(0, "0");
+            }
+            Ok(Hex(Vec::from_hex(&s).map_err(|err| {
+                serde::de::Error::custom(err.to_string())
+            })?))
+        }
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct JsonProof {
+        pub header_data: String,
```

Comment:
https://github.com/aurora-is-near/fast-bridge-common/blob/master/src/lib.rs#L29

May be something like that will be work in that case?

Reply:
added in 09068dcb60cd3b4f6d254a5c021e84ab3d45ffaa